main/src/library/Channel/Channel.flix

```diff
@@ -0,0 +1,582 @@
+            case Channel(_, channelLock, _, _, _, _, _)   => channelLock
+        }
+
+    pub def getBufferSize(c: Channel[a]): Int = match c {
+            case Channel(_, _, bufferSize, _, _, _, _)   => bufferSize
+        }
+
+    pub def getUnbuffered(c: Channel[a]): Bool = match c {
+            case Channel(_, _, _, unbuffered, _, _, _)   => unbuffered
+        }
+
+    pub def getElementQueue(c: Channel[a]): MutList[a] = match c {
+            case Channel(_, _, _, _, elementQueue, _, _)   => elementQueue
+        }
+
+    // TODO : make it a MutList or MutSet??
+    pub def getWaitingGetters(c: Channel[a]): MutList[LockConditionPair] = match c {
+            case Channel(_, _, _, _, _, waitingGetters, _)   => waitingGetters
+        }
+
+    pub def getWaitingSetters(c: Channel[a]): Condition = match c {
+            case Channel(_, _, _, _, _, _, waitingSetters)   => waitingSetters
+        }
+
+
+    /*
+    instance Order[Channel] {
+        pub def compare(x: Channel, y: Channel): Comparison =
+            compare(x.id, y.id)
+    }
+    */
+
+    /// instance Eq[Channel] {
+    ///     pub def eq(x: Channel[a], y: Channel[b]): Bool =
+    ///         channelId(x) == channelId(y)
+    /// }
+
+    /// def channelId(channel: Channel[a]): Int =
+    ///     let Channel(c) = channel;
+    ///     getId(c)
+
+    type alias ReentrantLock = ##java.util.concurrent.locks.ReentrantLock
+
+    /// type alias Lock = ##java.util.concurrent.locks.ReentrantLock
+
+    type alias LockConditionPair = (ReentrantLock, Condition)
+
+    pub def getLock(pair: LockConditionPair): ReentrantLock = match pair {
+            case (lock, _)   => lock
+        }
+
+    pub def getCondition(pair: LockConditionPair): Condition = match pair {
+            case (_, cond)   => cond
+        }
+
+    def java_signalAll(cond: Condition): Unit & Impure =
+        import java.util.concurrent.locks.Condition.signalAll() as signalAll;
+        signalAll(cond);
+        ()
+
+    def java_lock(reentrantLock: ReentrantLock): Unit & Impure =
+        import java.util.concurrent.locks.ReentrantLock.lock() as lock;
+        lock(reentrantLock);
+        ()
+
+    def java_unlock(reentrantLock: ReentrantLock): Unit & Impure =
+        import java.util.concurrent.locks.ReentrantLock.unlock() as unlock;
+        unlock(reentrantLock);
+        ()
+
+    def java_await(condition: Condition): Unit & Impure =
+        import java.util.concurrent.locks.Condition.await();
+        condition.await()
+
+    def newReentrantLock(): ReentrantLock & Impure =
+        import new java.util.concurrent.locks.ReentrantLock() as javaNewReentrantLock;
+        javaNewReentrantLock()
+
+    type alias Condition = ##java.util.concurrent.locks.Condition
+
+    def newCondition(reentrantLock: ReentrantLock): Condition & Impure =
+        import java.util.concurrent.locks.ReentrantLock.newCondition() as javaNewCondition;
+        javaNewCondition(reentrantLock)
+
```

Comment:
All of the above helper functions could be put into their own namespace, but we can talk about that later.